server/test/views/admin/programs/ProgramBlockViewTest.java

```diff
@@ -0,0 +1,157 @@
+package views.admin.programs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import j2html.tags.specialized.DivTag;
+import junitparams.JUnitParamsRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import services.applicant.question.Scalar;
+import services.program.predicate.AndNode;
+import services.program.predicate.LeafOperationExpressionNode;
+import services.program.predicate.Operator;
+import services.program.predicate.OrNode;
+import services.program.predicate.PredicateAction;
+import services.program.predicate.PredicateDefinition;
+import services.program.predicate.PredicateExpressionNode;
+import services.program.predicate.PredicateValue;
+import services.question.types.QuestionDefinition;
+import support.CfTestHelpers;
+import support.TestQuestionBank;
+
+@RunWith(JUnitParamsRunner.class)
+public class ProgramBlockViewTest {
+
+  private static final String BLOCK_NAME = "Block_name";
+  private TestQuestionBank testQuestionBank = new TestQuestionBank(/* canSave= */ false);
+  private ImmutableList<QuestionDefinition> questionDefinitions =
+      ImmutableList.of(
+          testQuestionBank.applicantDate().getQuestionDefinition(),
+          testQuestionBank.applicantEmail().getQuestionDefinition());
+
+  @Test
+  public void renderExistingPredicate_singleQuestion() {
+    var predicateDefinition =
+        PredicateDefinition.create(
+            PredicateExpressionNode.create(
+                LeafOperationExpressionNode.builder()
+                    .setQuestionId(testQuestionBank.applicantDate().id)
+                    .setScalar(Scalar.DATE)
+                    .setOperator(Operator.EQUAL_TO)
+                    .setComparedValue(CfTestHelpers.stringToPredicateDate("2023-01-01"))
+                    .build()),
+            PredicateAction.HIDE_BLOCK);
+
+    DivTag result =
+        new ProgramBlockViewTestChild()
+            .renderExistingPredicate(BLOCK_NAME, predicateDefinition, questionDefinitions);
+
+    assertThat(result.render())
+        .isEqualTo(
+            "<div>Block_name is hidden if &quot;applicant birth date&quot; date is equal to"
+                + " 2023-01-01</div>");
+  }
+
+  @Test
+  public void renderExistingPredicate_orOfSingleLayerAnds_singleAnd() {
+    var predicateDefinition =
+        PredicateDefinition.create(
+            PredicateExpressionNode.create(
+                OrNode.create(
+                    ImmutableList.of(
+                        PredicateExpressionNode.create(
+                            AndNode.create(
+                                ImmutableList.of(
+                                    PredicateExpressionNode.create(
```

Comment:
Optional: You could make this and the second and not in the below test a class const this would make this setup smaller and reuse the first Node at least.

Reply:
Personal preference is that it's easier to read when the node definitions are adjacent